### PR TITLE
doc: Revamped Resource Browser Doc

### PR DESCRIPTION
**Pull request by ashokdevtron**
**Done**
- Repurposed the old content and restructured the whole doc
- Shifted node operations from another doc to this
- Removed old cluster doc
- Added gifs wherever possible
- Added videos
- Moved a portion of ephemeral container to this doc
- Added glossary item for 'K8s object'
- Included a guide to download pod log 
- Added workload filter and last restart pod log
- Added redirection from old cluster doc to RB doc
- Replaced snapshots of node operations
- Added creatives for easy explanation of node ops
- RBAC
- Proofreading (for grammar and errors)

### CODE REVIEW

File: docs/user-guide/resource-browser.md [hunk 1]
```diff
@@ -1,71 +1,438 @@
...
-### Create Kubernetes Resource
+{% embed url="https://www.youtube.com/watch?v=E-V-ELCXtfs" caption="Filtering Workloads in Devtron" %}
 
-**Note**: As a pre-requisite, you must have a basic understanding of Kubernetes Cluster, Resources, Kinds.
+### Edit Manifest 
 
-You can create a Kubernetes resource by passing definition YAML file. You can create more than one resource by separating the resource YAMLs by ‘---’.
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have super-admin permission to edit a manifest.
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:83]: I think an admin of a resource should be able to edit the manifest @uxarya-d ?

File: docs/user-guide/resource-browser.md [hunk 2]
```diff
@@ -1,71 +1,438 @@
...
+
+You can monitor activities like creation, deletion, updation, scaling, or errors in the resources involved.
+
+![Figure 6: Viewing All Events](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/kubernetes-resource-browser/events.gif)
+
+### Delete
+
+{% hint style="warning" %}
+### Who Can Perform This Action?
+Users need to have super-admin permission to delete a resource.
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:100]: same need to validate all RBACs

File: docs/user-guide/resource-browser.md [hunk 3]
```diff
@@ -1,71 +1,438 @@
...
+{% endhint %}
+
+You can delete an unwanted resource if it is orphaned and no longer required by your applications.
+
+![Figure 7: Deleting a Resource](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/kubernetes-resource-browser/delete.gif)
+
+You can perform all the actions on the following:
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:107]: Why mention this? User can do on all kubernetes objects...

File: docs/user-guide/resource-browser.md [hunk 4]
```diff
@@ -1,71 +1,438 @@
 # Resource Browser
 
-`Resource Browser` lists all of the resources running in your cluster. You can use it to view, inspect, manage, and delete resources in your cluster. You can also create resources from the `Resource Browser`.
+## Introduction
 
-Resource Browser are helpful for troubleshooting issues. It supports multi-cluster.
+The Devtron Resource Browser provides you a central interface to view and manage all your [Kubernetes objects](../../reference/glossary.md#objects) across clusters.  It helps you perform key actions like viewing logs, editing live manifests, and even creating/deleting resources directly from the user interface. This is especially useful for troubleshooting purposes as it supports multi-cluster too.
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:5]: We have a few good blogs on resource browser, should we link it here somewhere? Why should one use resource browser over kubectl type blogs?

File: docs/user-guide/resource-browser.md [hunk 5]
```diff
@@ -1,71 +1,438 @@
...
+
+The components on a typical node include the `kubelet`, a `container runtime`, and the `kube-proxy`.
+
+If you have multiple nodes, you can search a node by name or label in the search bar. The search result will display the following information of the node. To display a parameter of a node, use `Columns` on the right side, select the parameter to display from the drop-down list, and click **Apply**.
+
+![Figure 8: Searching and Filtering Nodes](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/kubernetes-resource-browser/cluster-nodes.jpg)
+
+| Fields | Description |
+| --- | --- |
+| Node | User-defined name for the node in Devtron. E.g. `demo-new`.<br>Note: Two nodes cannot have the same name at the same time.</br> |
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:126]: How can a user define name of a node on devtron?

File: docs/user-guide/resource-browser.md [hunk 6]
```diff
@@ -1,71 +1,438 @@
...
+
+  ![Figure 13c: Adding Taints](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/kubernetes-resource-browser/taint-dialog.jpg)
+
+* Click **Save**.
+
+You can also add more taints using **+ Add taint button**, or delete the existing taint by using the delete icon.
+
+#### Taint Effects
+
+A taint can produce three possible effects:
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:241]: We should ideally link such conceptual things to k8s official docs. and in our doc we should focus on how to's...

File: docs/user-guide/resource-browser.md [hunk 7]
```diff
@@ -1,71 +1,438 @@
...
+## Cluster Terminal
+
+User with [super-admin](./global-configurations/authorization/user-access.md#assign-super-admin-permission) access can now troubleshoot cluster issues by accessing the cluster terminal from Devtron. You can select an image from the list that has all CLI utilities like kubectl, helm, netshoot etc. or can use a custom image, which is publicly available.
+
+To troubleshoot a cluster or a specific node in a cluster, click the terminal icon on the right side.
+
+![Figure 19: Terminal Icon](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/kubernetes-resource-browser/cluster-terminal.gif)
+
+* You will see the user-defined name for the cluster in Devtron. E.g. `default-cluster`.
+* Select the node you wish to troubleshoot from the `Node` drop-down. E.g. `demo-new`.
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:374]: We have auto select too

File: docs/user-guide/resource-browser.md [hunk 8]
```diff
@@ -1,71 +1,438 @@
...
+
+### Logs
+
+Examining your cluster's pods helps you understand the health of your application. By inspecting pod logs, you can check the performance and identify if there are any failures. This is especially useful for debugging any issues effectively.
+
+Moreover, you can download the pod logs for ease of sharing and troubleshooting as shown in the below video.
+
+{% embed url="https://www.youtube.com/watch?v=PP0ZKAZCT58" caption="Downloading Pod Logs" %}
+
+#### Last Restart Pod Log
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:301]: It's not just logs...

File: docs/user-guide/resource-browser.md [hunk 9]
```diff
@@ -1,71 +1,438 @@
...
+
+```bash
+kubectl describe pod <podname>
+```
+
+| Pod Status       | Description                             | Common Causes                              |
+| ---------------- | --------------------------------------- | ------------------------------------------ |
+| Pending          | Pod cannot be scheduled onto a node     | Insufficient resources (CPU, memory, etc.) |
+| Waiting          | Pod has been scheduled but cannot run   | Failed to pull container image             |
+| CrashLoopBackOff | Container keeps crashing and restarting | Incorrect access token for API interaction |
```
Comment (abhibhaw) [inline docs/user-guide/resource-browser.md:401]: Is this example pod statuses or all pod statuses? [Ref](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/)

**Review by abhibhaw — CHANGES_REQUESTED** [review]
- RBAC needs to be crosschecked, current mentions concludes, resource browser can only be used by superadmins. (I believe RBAC is pending as per PR description)
- We should attach more reference links to k8s docs

**Comment by abhibhaw** [comment]
Request someone from @devtron-labs/devops-team to review this doc.

File: docs/user-guide/resource-browser.md [hunk 10]
```diff
@@ -41,16 +41,21 @@ This shows errors in the cluster. If no error is present in the cluster, Resourc
 
 ### Catalog Framework
 
-Based on the schema provided in the catalog framework, you can add relevant details for each cluster. Refer [Catalog Framework](./global-configurations/catalog-framework.md) for more details.
+Based on the schema provided in the catalog framework, you can add relevant details for each cluster. Refer [Catalog Framework](./global-configurations/catalog-framework.md) for more details. Users need to have super-admin permission to edit the catalog framework.
 
 ### Readme
 
-You can also include additional information about your cluster using the Markdown editor.
+You can also include additional information about your cluster using the Markdown editor. You will need superadmin permission to edit it. Users need to have super-admin permission to edit the readme file.
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md]: Seems to be in an older commit. Fixed in latest commit.

File: docs/user-guide/resource-browser.md (hunk 1 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:83]: You are right, @abhibhaw. 
Now I checked with editing a label as an admin, and it worked. 

Initially, my inference was based on this error which led me to believe it cannot be edited by anyone other than superadmin (gif shared below). 
409 error

File: docs/user-guide/resource-browser.md (hunk 5 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:126]: Fixing it. Got leaked from our existing doc, I suppose (snapshot attached below)

image

File: docs/user-guide/resource-browser.md (hunk 3 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:107]: Removed

File: docs/user-guide/resource-browser.md (hunk 9 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:401]: Those were example pod statuses. But, I get what you're saying, the list of statuses and conditions is exhaustive. 
Let me just remove the statuses and point them to the K8s reference instead.

File: docs/user-guide/resource-browser.md (hunk 4 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:5]: Added

File: docs/user-guide/resource-browser.md [hunk 11]
```diff
@@ -1,71 +1,438 @@
...
+{% endhint %}
 
-An example that shows the required fields and object specifications for a Kubernetes Deployment:
+You can edit the [manifest](../reference/glossary.md#manifest) of a Kubernetes object. This can be for fixing errors, scaling resources, or changing configuration.
+
+![Figure 5: Editing a Live Manifest](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/kubernetes-resource-browser/edit-live-manifest.gif)
+
+### View Events
+
+You can monitor activities like creation, deletion, updation, scaling, or errors in the resources involved.
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:92]: Need more clarity on this, please

File: docs/user-guide/resource-browser.md (hunk 2 above)
Comment (ashokdevtron) [reply docs/user-guide/resource-browser.md:100]: Done, tested from my side using another user account.

**Comment by ashokdevtron** [comment]
> Request someone from @devtron-labs/devops-team to review this doc.

Reviewed by @ajaydevtron on May 10th

**Review by abhibhaw — APPROVED** [review]
LGTM

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**